File: src/frontend/src/types/grados.ts
export interface Grado {
  id: number;
  codigo: string;
  nombre: string;
  facultad: string;
}

export interface CrearGradoRequest {
  codigo: string;
  nombre: string;
  facultad: string;
}

export interface EditarGradoRequest {
  nombre?: string;
  facultad?: string;
}

File: src/backend/app/schemas/grados.py
from pydantic import BaseModel, ConfigDict


class GradoOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    codigo: str
    nombre: str
    facultad: str


class CrearGradoRequest(BaseModel):
    codigo: str
    nombre: str
    facultad: str


class EditarGradoRequest(BaseModel):
    """`codigo` no editable post-creación; si llega se descarta."""

    model_config = ConfigDict(extra="ignore")

    nombre: str | None = None
    facultad: str | None = None
